Add Faterium Milestone 2

enfipy opened the pull request:
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [x] This pull request is being made by the same account as the accepted application.
- [x] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/milestone-deliverables-guidelines.md).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/1264.

## Context

The main goal of the Milestone 2 was to create UI design (see [wireframes](https://github.com/w3f/Grants-Program/blob/master/applications/faterium.md#wireframes) for the reference) and all main functionality of Faterium Platform: Communities, User Profiles, Categories, Assets, Creation Pages, Home Page, etc.

https://user-images.githubusercontent.com/24860875/214779778-fd344725-737c-4f46-9f99-db6676ae3c37.mp4

Also, as specified in milestone delivery - we published an article on our Medium publication [here](https://medium.com/@dodorare/power-of-voting-why-does-the-world-need-a-universal-voting-platform-e98cf64e06c9).

0xCaso reviewed (commented):
Hi @enfipy, thanks a lot for the new delivery. I'm starting the evaluation, I'll get back to you the next week!

0xCaso reviewed (changes requested):
Here I am @enfipy, I've just posted my evaluation [here](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/faterium_2_0xCaso.md). If you have any questions let me know!

0xCaso commented:
Note: please ignore in **0c. Testing Guide** the fact that tests do not work, because they did work in another machine. Probably I have some configuration problems in that one.

enfipy commented:
Hey @0xCaso, glad to see you!

Thanks a lot for your evaluation and questions. I fixed all issues you mentioned (except those errors as it was hard to determine the error origin):
1. **Added categories creation**;
2. **Improved docker** (thanks for mentioning the docker-compose ports issue, I fixed it and added the same file to the `faterium-dapp` repository to enhance developer experience);
3. **Better error handling** (now errors look better for PocketBase issues and create a link for a specific block for Substrate errors);
4. **Tests** (now it's possible to run against `testnet` and `local`);
5. **Issue with minimum balance** (I fixed it by checking the correct amount on the frontend side because I think it should be more than 0 by design).

Now about the question regarding the `collections.json` file - it's sufficient to update only this file as PocketBase and `faterium-server` [pull](https://github.com/faterium/faterium-server/blob/v0.2.1/cmd/main.go#L56) this file during the [Docker image](https://github.com/faterium/faterium-server/blob/v0.2.1/Dockerfile#L18) build. But it is also important to download the [latest docker image for the server](https://github.com/faterium/faterium-server/pkgs/container/faterium-server/66974895?tag=latest).

Let me now if you have any other questions or issues, thanks!

0xCaso commented:
Thank you for the update @enfipy, and thanks for the answers, all clear!

By the way, I pulled the `faterium-server` and `faterium-dapp` repos, and ran everything, but somehow seems like the server doesn't work (I don't see the images, and if I try to create something that needs an image, it gives error). 
Could it be a problem with the IPFS instance?

enfipy commented:
Hi @0xCaso, sorry I can't reproduce the error - can you provide some more context? And can you please make sure that you use the latest docker images and cleared the cache? Because when I run the local setup with docker and frontend - everything is working as expected.

0xCaso commented:
Yea probably I'm missing something.

I deleted the containers, the images, ran `docker system prune -a`. Also deleted and re-cloned the repos. Then:
- I ran docker-compose up in the server repo;
- I started the frontend.

I'm doing this in my local machine so it's not a ports-related issue.

This happens when I try to create a profile:

![Screenshot 2023-01-31 at 14 41 06](https://user-images.githubusercontent.com/22419450/215776240-7c9cd0b4-5ed7-420d-bb95-887c789f4739.png)

or a community:

![Screenshot 2023-01-31 at 14 30 53](https://user-images.githubusercontent.com/22419450/215776297-833628cf-b7b7-4999-b809-b13e901b19c6.png)

I manage to create a poll, but it shows like this (an image is missing, note also the profile pic has the same issue):

![Screenshot 2023-01-31 at 14 32 09](https://user-images.githubusercontent.com/22419450/215776452-b863fc88-ca0b-4577-846c-1bc5acd6a2aa.png)

enfipy commented:
Oh, this looks like definitely problem with cache. If you open developer tools, then Application, and then click clear cache - the problem disappears.

I will push a script that will check it and delete problematic cookies automatically.

enfipy commented:
@0xCaso Thank you for pointing out, just pushed a fix that should resolve the issue :)

0xCaso reviewed (changes requested):
Thanks again @enfipy for the update, and yes the problem was the browser cache, great!

I updated the evaluation [here](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/faterium_2_0xCaso.md), I left 3 other minor comments.

enfipy commented:
Hey @0xCaso, thank you for pointing out those flows (I somehow forgot to add them).

I fixed the problems you mentioned:
1. Now it's possible to link the community to a category (and also, now when you open a specific category - you will see Communities instead of Polls);
2. Instead, I made it impossible to create a poll without community as this is "how it's supposed to be" (you should get an error if you skip selection from dropdown);
3. Fixed it.

enfipy commented:
@0xCaso Fixed 👍

0xCaso reviewed (approved):
Thanks a lot, I'm accepting the delivery and sharing it with the team!

0xCaso commented:
The grant is completed, congrats @enfipy and thanks for the cooperation!

I've forwarded internally the invoice, the payment will be done within 14 days (could be before).

enfipy commented:
Thanks for the cooperation to you too @0xCaso!
